prototype/FFT_image_processing/serial_port.c

```diff
@@ -0,0 +1,316 @@
+// C library headers
+#include "serial_port.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Linux headers
+#include <errno.h>   // Error integer and strerror() function
+#include <fcntl.h>   // Contains file controls like O_RDWR
+#include <termios.h> // Contains POSIX terminal control definitions
+#include <unistd.h>  // write(), read(), close()
+
+static int setupSerialPort(int serialPort, const speed_t baudRate);
+static speed_t getBaudRateCode(const unsigned baudRate);
+static int fileDescriptor = -1;
```

Comment:
What happens if we need 2 serial connections because 2 arduinos are connected on the same computer. Global variables won't allow you to do that. I think we should wrap both fileDescriptor and serialWriteBufferSize and implement a logic similar to pthread. 
```
pthread_t thread;
pthread_create(&thread1, NULL, func, NULL);
pthread_join(&thread, NULL);
```

This will allow us to use as many serial connections as we want at the same time. What do you think ?